src/game/Commands/Result.ts

```diff
@@ -0,0 +1,29 @@
+export interface Ok<A> {
+  readonly _kind: 'Ok',
+  value: A
+}
+
+export const ok = (value: any): Ok<any> => ({
+  _kind: 'Ok',
+  value
+})
+
+export interface Err<E> {
+  readonly _kind: 'Err',
+  value: E
+}
+
+export const err = (value: any): Err<any> => ({
+  _kind: 'Err',
+  value
+})
+
+export type Result<E, A>
+  = Ok<A>
+  | Err<E>
+
+export const isOk = (result: Result<any, any>): boolean => 
```

Comment:
Is this intended as a type guard? Would using `result is Ok` be better?
```suggestion
export const isOk = (result: Result<any, any>): result is Ok => 
```